README.md

```diff
@@ -144,4 +146,43 @@ with f:
     f.search(doc_query, parameters={'filter': filter})
 ```
 
+### Limit results
+
+In some cases, you will want to limit the total number of retrieved results. `ElasticSearchIndexer` uses the `limit` argument 
+from the `match` function to set this limit. Note that when using `shards=N`, the `limit=K` is the number of retrieved results for **each shard** and total number of retrieved results is `N*K`. By default, `limits` is set to `20`. For more information about shards, please read [Jina Documentation](https://docs.jina.ai/fundamentals/flow/topology/#partition-data-by-using-shards)
+
+```python
+f =  Flow().add(
+    uses='jinahub://ElasticSearchIndexer',
+    uses_with={'match_args': {'limit': 10}})
+```
+
+### Configure other search behaviors
+
+You can use `match_args` argument to pass arguments to the `match` function as below. The match function will be called
+during `/search` endpoint.
+
+```python
+f =  Flow().add(
+     uses='jinahub://ElasticSearchIndexer',
+     uses_with={
+         'match_args': {
+             'metric': 'euclidean',
+             'use_scipy': True}})
+```
+
+- For more details about overriding configurations, please refer to [here](https://docs.jina.ai/fundamentals/executor/executor-in-flow/#special-executor-attributes).
+- You can find more about the `match` function at [here](https://docarray.jina.ai/api/docarray.array.mixins.match/#docarray.array.mixins.match.MatchMixin.match)
+
+### Configure the Search Behaviors on-the-fly
+
+**At search time**, you can also pass arguments to config the `match` function. This can be useful when users want to query with different arguments for different data requests. For instance, the following codes query with a custom `limit` in `parameters` and only retrieve the top 100 nearest neighbors. This will overwrite existing `match_args` if defined when initialized
```

Comment:
```suggestion
**At search time**, you can also pass arguments to config the `match` function. This can be useful when users want to query with different arguments for different data requests. For instance, the following codes query with a custom `limit` in `parameters` and only retrieve the top 100 nearest neighbors. This will override existing `match_args` if defined during Executor initialization.
```